cpp/include/raft/sparse/hierarchy/detail/mst.cuh

```diff
@@ -116,7 +116,7 @@ void connect_knn_graph(const raft::handle_t &handle, const value_t *X,
 
   // On the second call, we hand the MST the original colors
   // and the new set of edges and let it restart the optimization process
-  auto new_mst = raft::mst::mst<value_idx, value_idx, value_t>(
+  auto new_mst = raft::mst::mst<value_idx, value_idx, value_t, double>(
```

Comment:
Using 64b is only helpful if `max_gap` is large compared to `max_weight`. Always using 64b may unnecessarily slow down the execution.

Reply:
This is specifically to enable HDBSCAN (and remove errors in SLHC). While dynamic weight selection may help, MST is far from being the bottleneck for HDBSCAN, and we are trying to make HDBSCAN for this release so it's the path of least resistance for now